### DISCUSSION TITLE
Separate downloaded web apps from the browser it was downloaded in

### DISCUSSION THREAD

**Discussion by Harmxn02** [File-New-Project/EarTrumpet#1258] [Ideas]
downloadable web apps's  volume is combined with the browser in which they were installed, instead of being its own volume-controllable program

### What I mean with Downloadable web app

Some websites allow you to download the web-app version of the website.  For example youtube.com or open.spotify.com if you press the icon, it will install YouTube or Spotify as a separate app

![image](https://user-images.githubusercontent.com/64677817/199047034-b5ec2ff9-4482-4bd6-ae31-06e96b016247.png)
![image](https://user-images.githubusercontent.com/64677817/199047437-92c17511-f66d-4dcc-b6df-a758a8ad11d0.png)

### The issue 

If you want to volume control that app (for example Spotify which I installed using MS Edge) this is what is shown:

![image](https://user-images.githubusercontent.com/64677817/199048988-d0f48059-402c-49d3-af7a-7d284d673a1b.png)

Instead of it having a seperate icon for MS Edge and another for the downloaded web app, they're combined into 1. So when you lower the volume of EDGE (perhaps a certain Reddit video is too loud), it also lowers the Spotify volume.

**_You are not able to control the music without going into Spotify itself, instead of using this app functionality_**

### Work-around

Download those web apps in a different browser. Then you will have 2 seperate programs. But then the issue is that the web app downloaded in a different browser, will have the icon of the browser and not the icon of the app

![image](https://user-images.githubusercontent.com/64677817/199050135-39e46885-6b63-4a5c-b5a6-821622bf5f49.png)

### Real solution

Download those apps from the Windows Store. But not all of those apps have a version in the Windows Store

**Comment by riverar** [comment]
Also related https://github.com/File-New-Project/EarTrumpet/discussions/704.

As of today, neither Edge/Windows support this. Part of the reason is that audio in Chrome/Edge is _brokered_, that is piped into a single process in charge of emitting audio, with no API to differentiate between streams. (This is typically the first tab/browser process started.)

**Reply by Harmxn02** [reply]
That's a bummer. I guess I'll just change Chrome's icon.

**Reply by riverar** [reply]
Yeah, it's extremely frustrating. Sometimes it feels like we're going backwards in terms of audio on Windows!
